components/Textarea/Textarea.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+'use client';
+import { cn } from '@/utils/utils';
+import { useImperativeHandle } from 'react';
+import * as React from 'react';
+
+interface UseAutosizeTextAreaProps {
+  textAreaRef: HTMLTextAreaElement | null;
+  minHeight?: number;
+  maxHeight?: number;
+  triggerAutoSize: string;
+}
+
+/**
+ * useAutosizeTextArea hook
+ * @param {UseAutosizeTextAreaProps} props - The props
+ */
+export const useAutosizeTextArea = ({
+  textAreaRef,
+  triggerAutoSize,
+  maxHeight = Number.MAX_SAFE_INTEGER,
+  minHeight = 0,
+}: UseAutosizeTextAreaProps): void => {
+  const [init, setInit] = React.useState(true);
+  React.useEffect(() => {
+    const offsetBorder = 2;
+    if (textAreaRef) {
+      if (init) {
+        textAreaRef.style.minHeight = `${minHeight + offsetBorder}px`;
+        if (maxHeight > minHeight) {
+          textAreaRef.style.maxHeight = `${maxHeight}px`;
+        }
```

Comment:
Are these if statement nestings necessary, or are there ways we can combine some of them into one if conditional?

Reply:
This would be a question for Ryan. These were his changes made that got merged into my branch after approval.